### ISSUE TITLE
`RsImplItem` and `RsTraitItem` should be the same PsiElement (or siblings)

### ISSUE THREAD

**Issue by matklad** [intellij-rust/intellij-rust#1294]
`Trait`s and `Impl`s are very alike: they are collections of functions. And in generic context, sometimes you know that type `X` implements some `T` via an impl, and sometimes without an impl (for example, if `X` is a type parameter, and you have `X: T` bound).

So it makes sense to coalesce this now two completely different entities into a single thing and change horrendous 

```
fun findImplsAndTraits(project: Project, ty: Ty): Pair<Collection<BoundElement<RsImplItem>>, Collection<BoundElement<RsTraitItem>>> 
```

to a nicer

```
fun findStuff(project: Project, ty: Ty): Collection<BoundElement<RsStuff>>, {
``` 

As usual, the most hard thing is naming...

**Comment by Undin** [comment]
Apparently, it's closed by #1387

**Comment by matklad** [comment]
Yep!
